src/Stratis.Bitcoin.Api.Tests/ApiSettingsTest.cs

```diff
@@ -29,15 +30,14 @@ public void GivenNoApiSettingsAreProvided_AndOnBitcoinNetwork_ThenDefaultSetting
         {
             // Arrange.
             Network network = Network.Main;
-            NodeSettings nodeSettings = new NodeSettings(network).LoadArguments(new string[0]);
+            NodeSettings nodeSettings = new NodeSettings(network, args:new[] { $"-datadir={CreateTestDir(this)}" });
```

Comment:
@quantumagi I checked out your branch and run soime of these tests without passing in:

```args:new[] { $"-datadir={CreateTestDir(this)}" ``` 

and they still passed?

Can you please check if its needed everywhere?

Reply:
This is to prevent the data dir from defaulting to a folder relative to the current user which is not a valid folder for testing purposes.